api/src/main/java/org/pmiops/workbench/google/CloudStorageService.java

```diff
@@ -56,4 +57,14 @@ ServiceAccountCredentials getGarbageCollectionServiceAccountCredentials(
   String getMoodleApiKey();
 
   String getCaptchaServerKey();
+
+  default FileDetail blobToFileDetail(Blob blob, String bucketName) {
+    String[] parts = blob.getName().split("/");
+    FileDetail fileDetail = new FileDetail();
+    fileDetail.setName(parts[parts.length - 1]);
+    fileDetail.setPath("gs://" + bucketName + "/" + blob.getName());
+    fileDetail.setLastModifiedTime(blob.getUpdateTime());
+    fileDetail.setSize(blob.getSize());
```

Comment:
This line is the only addition.  The rest was moved as-is from NotebooksServiceImpl.